internal/stringutil/util.go

```diff
@@ -2,6 +2,7 @@
 package stringutil
 
 import (
+	"net/url"
```

Comment:
I am suspicious of using this for anything for JS stuff; can we stick a `!!!` onto the uses of this so we can say "come back and see if we should change this"? By using it we're somewhat at the whim of what Go does here, which could be good but could be bad.

Reply:
This should be equivalent to `encodeURIComponent`. I went through and compared the encoding rules Go uses to the ones JS uses, and these two mechanisms are equivalent. However, Go has no equivalent to `encodeURI`, which is why I implemented it myself.